cookbook/sections/data-publishers.adoc

```diff
@@ -82,9 +82,137 @@ Note:
 
 Of course, always ensure your WNM is valid (see <<Validate a WIS2 Notification Message>> for more information).
 
-=== Validate a WMO Core Metadata Profile record
+=== Publishing a WIS2 Notification Message with embedded data
 
-The [pywcmp](https://github.com/wmo-im/pywcmp) tool provides a test suite to validate a message against the WCMP2 specification requirements, as well as a Python API for application integration.  Consult the pywcmp README on GitHub for more information/examples.
+A https://wmo-im.github.io/wis2-notification-message[WIS2 Notification Message] provides the ability to embed data as part of a message (see the https://wmo-im.github.io/wis2-notification-message/standard/wis2-notification-message-STABLE.html#_1_15_properties_content[Properties / Content] section of the specification).  A data publisher may wish to provide embedded data for direct access, thereby skipping the need for accessing the data via the notification link (see the https://wmo-im.github.io/wis2-notification-message/standard/wis2-notification-message-STABLE.html#_1_16_links[Links] section of the specification).  A data publisher also may not wish to manage or publish the data link to infrastructure to an HTTP or FTP server.
+
+The WIS2 Notification Message requires at least one link object for access to data via HTTP, HTTPS, FTP, or SFTP.  In the context of core data, WIS2 Global Cache services use the link object to download data for resilient data access by WIS2 data consumers.
+
+For of publishing embedded core data:
+
+- use `properties.content` object to provide the data
+- use `cache=False` to instruct WIS2 Global Cache Services to not cache the data
+- provide a link object whose `href` results in HTTP 204 (No Content)
+
+To provide an HTTP 204 link, there a couple of options:
+
+**Option 1: configure your web server to provide an endpoint that always returns 204**:
+
+[source,console]
+----
+# always return HTTP 204 when accessing /no-content-here
+location /no-content-here/ {
+  return 204;
+}
+----
+
+which then would result in the link object as follows:
+
+[source,json]
+----
+{
+    "href": "https://example.org/no-content-here"
+}
+----
+
+**Option 2: use a service that provides HTTP responses as a service**.  One such service is the https://http.codes[HTTP Response API], which then would result in the link object as follows:
+
+[source,json]
+----
+{
+    "href": "https://http.codes/204"
+}
+----
+
+.Embedded data example for core data:
```

Comment:
If we say specifically core data, we should provide an example for recommended.
Or just call it 
. Embedded data example